### PR TITLE
Add `ValidatedNotNullAttribute` and corresponding tests

### LINKED ISSUE #1: Argument methods require [ValidatedNotNull]
Background: https://github.com/OnTopicCMS/OnTopic-Library/commit/b9e0151d8bacd67a49214054fea9a632167676a8

Resolves: https://github.com/ashmind/SqlFake/pull/2

Suggestion: Upgrade Argument to .netstandard 2.1 and add [ValidatedNotNull] to parameters

### PR DESCRIPTION
**Pull request by ashmind**
Fixes #1.

### CODE REVIEW

File: Argument/Argument.csproj [hunk 1]
```diff
@@ -5,9 +5,9 @@ this.user = Argument.NotNull("user", user).
 
 Less fancy than approaches based on lambdas/anonymous classes, but less scary performance-wise (given that argument checks are often all over the place).
 
-Includes Code Contracts [ContractArgumentValidator] annotation (.NET 4.5 only), ReSharper annotations and C# 8 nullability annotations.</Description>
-    <Copyright>Copyright (c) 2013–2019 Andrey Shchekin</Copyright>
-    <VersionPrefix>1.1.1</VersionPrefix>
+Annotations: C# 8 nullable reference types, ReSharper, Microsoft.CodeQuality.Analyzers, Code Contracts (.NET 4.5 only).</Description>
+    <Copyright>Copyright (c) 2013–2020 Andrey Shchekin</Copyright>
+    <VersionPrefix>1.1.2</VersionPrefix>
```
Comment (ashmind) [inline Argument/Argument.csproj]: Should probably be 1.2 as it is a new feature that does affect the API...

File: Argument/Internal/ValidatedNotNullAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+using System;
+
+namespace ArgumentInternal {
```
Comment (jsk95) [inline Argument/Internal/ValidatedNotNullAttribute.cs:3]: Wouldn't the world be better with a dot (I feel a no coming though)

**Review by jsk95 — APPROVED** [review]
LGTM 👍

File: Argument/Internal/ValidatedNotNullAttribute.cs (hunk 2 above)
Comment (ashmind) [reply Argument/Internal/ValidatedNotNullAttribute.cs:3]: Can't do it -- will conflict with `class Argument` which has no namespace.

Comment (jsk95) [reply Argument/Internal/ValidatedNotNullAttribute.cs:3]: Ah fair enough

**Pull request merged**